src/v/hashing/secure.h

```diff
@@ -25,6 +25,17 @@ class hmac_exception : public std::exception {
     const char* _msg;
 };
 
+class hash_exception : public std::exception {
```

Comment:
We have two different exception types for hashing and hmac. Maybe we can use one type for both, since they both use the same gnutls error codes? I also use gnutls in S3 lib and there I have an error_category that works with std::error_code. Maybe we can generalize and move it into some common location. The current approach is also OK to me, TBH.